File: 791/CustomSortString.cs
/*
https://leetcode.com/problems/custom-sort-string/description/?envType=daily-question&envId=2024-03-11
*/

public class Solution {
    public string CustomSortString(string order, string s) {
        int szordr=order.Length, szs=s.Length;
        HashSet<int>seenidxs=new HashSet<int>();
        string res="";
        for(int i=0; i<szordr; ++i){
            for(int j=0; j<szs; ++j){
                if(!seenidxs.Contains(j)){
                    if(s[j]==order[i]){
                        res+=order[i];
                        seenidxs.Add(j);
                    }
                }
            }
        }
        for(int i=0; i<szs; ++i){
            if(!seenidxs.Contains(i)){
                res+=s[i];
            }
        }
        return res;
    }
}

File: 791/CustomSortString.js
/*
https://leetcode.com/problems/custom-sort-string/description/?envType=daily-question&envId=2024-03-11
*/

/**
 * @param {string} order
 * @param {string} s
 * @return {string}
 */
var customSortString = function(order, s) {
    let szordr=order.length, szs=s.length;
    let seenidxs=new Set();
    let res="";
    for(let i=0; i<szordr; ++i){
        for(let j=0; j<szs; ++j){
            if(!seenidxs.has(j)){
                if(s[j]===order[i]){
                    res+=order[i];
                    seenidxs.add(j);
                }
            }
        }
    }
    for(let i=0; i<szs; ++i){
        if(!seenidxs.has(i)){
            res+=s[i];
        }
    }
    return res;
};